### PR TITLE
Add  OOP school library classes

### PR DESCRIPTION
**Pull request by ismayito**
### School library classes
_______________________________________________________________________________________________________
#### Summary 
_______________________________________________________________________________________________________
In this I added the following
- Created class Person with instance variables of id,name and age
- Created class Student that inherits from class Person 
- Added new method of play_hooky in class Student
- Created class Teacher  that inherits from class Person
- Overrided can_use_services? method from class Person to always return true

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Person
+  def initialize(name = 'unknown', age = nil, parent_permission: true)
```
Comment (DammyShittu) [inline person.rb]: - [ ] Age should not be an optional parameter. It should be mandated.

  ```
  def initialize(age, name = 'unknown', parent_permission: true)
  ```

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Student < Person
+  def initialize(classroom)
+    super('unknown', nil, parent_permission: true)
+    @classrom = classroom
+  end
```
Comment (DammyShittu) [inline student.rb]: - [ ] Kindly build your constructor well. You should require age, name and parent_permission and they should be well parsed into the `super` method.

  ```
  def initialize(classroom, age, name = 'unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
    @classroom = classroom
  end
  ```

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(specialization)
+    super('unknown', nil, parent_permission: true)
+    @specialization = specialization
+  end
```
Comment (DammyShittu) [inline teacher.rb]: - [ ] Kindly build your constructor well. You should require the necessary methods and they should be well parsed into the `super` method.

  ```
  def initialize(specialization, age, name = 'unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
    @specialization = specialization
  end
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @ismayito,

## Highlights
✔️ Private method feature

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @ismayito,

✔️ Implementation of classes in different files based on convention

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**